src/main/scala/it/pagopa/interop/agreementprocess/api/impl/ResponseHandlers.scala

```diff
@@ -0,0 +1,160 @@
+package it.pagopa.interop.agreementprocess.api.impl
+
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.LoggerTakingImplicit
+import it.pagopa.interop.agreementprocess.error.AgreementProcessErrors._
+import it.pagopa.interop.commons.logging.ContextFieldsToLog
+import it.pagopa.interop.commons.utils.errors.{AkkaResponses, ServiceCode}
+
+import scala.util.{Failure, Success, Try}
+
+object ResponseHandlers extends AkkaResponses {
+
+  implicit val serviceCode: ServiceCode = ServiceCode("005")
+
+  def createAgreementResponse[T](logMessage: String)(
+    success: T => Route
+  )(result: Try[T])(implicit contexts: Seq[(String, String)], logger: LoggerTakingImplicit[ContextFieldsToLog]): Route =
+    result match {
+      case Success(s)                                => success(s)
+      case Failure(ex: DescriptorNotInExpectedState) => badRequest(ex, logMessage)
+      case Failure(ex: AgreementAlreadyExists)       => badRequest(ex, logMessage)
```

Comment:
conflict maybe? apply this comment to every other `...alreadyExists` exception here